Tabs on Main page

Deman4ig opened the pull request:
1. Refactor main page to integrate tabs
2. Integrate tabs on the main page

Deman4ig committed:
Tabs on Main page

1. Refactor main page to integrate tabs
2. Integrate tabs on the main page

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

GSemikozov commented on components/main/main.css:
```diff
@@ -1,9 +1,16 @@
-    /*margin: 0;*/
-    padding: 43px 0;
+.main-page__header {
+    flex-grow: 0;
```
flex-grow по умолчанию auto = 0, поэтому явно указывать не обязательно)

Deman4ig replied:
да, действительно!

GSemikozov commented on components/tabs/tabs.css:
```diff
@@ -46,3 +46,91 @@
     display: none;
 }
+
+.tabs input[type=radio] {
```
тут и ниже - все селекторы тегов лучше заменить на классы по БЭМ и избегать использования вложенных селектор (без необходимости)

Deman4ig replied:
вообще думал перелопатить эти стили уже когда в SCSS все буду бомбить..:)

сделал вообщем где можно было классы а где то модификаторы, например `tabs__nav-label_tab0` вместо `.tabs__nav-item[for=tab0]`

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

Deman4ig committed:
Refactor tabs.css stylesheet

Deman4ig committed:
Merge branch 'master' of github.com:js-tasks-ru/htmlcss-20210513_deman4ig into lesson-05_mainpage-tabs

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

Deman4ig committed:
Refactor component pages

1. Reuse CSS tabs from the main page
2. Clean up template.css stylesheet
3. Clean up tabs.css stylesheet

Deman4ig committed:
Complete component page for custom form elements

1. Add component preview
2. Add code of the component
3. Fix minor issue with icon position

Deman4ig committed:
Insert custom form elements into the main page

Deman4ig committed:
Add links from the main page to component pages

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

GSemikozov commented on components/component-preview/component-preview.css:
```diff
@@ -10,6 +10,23 @@
+    flex-direction: column;
+}
+
+.component-preview__flex_gap-16px {
```
тут и ниже - слишком специфичные модификаторы. В контексте дизайн системы первый отступ скорее всего можно было бы обозначить как `.component-preview__flex_gap-sm`, а ниже - `.component-preview__flex_gap-md` как вариант

Deman4ig replied:
понял - принял

Deman4ig committed:
Refactor classes name in component-preview

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.
